src/map/zone_entities.cpp

```diff
@@ -1367,6 +1367,29 @@ void CZoneEntities::ZoneServer(time_point tick, bool check_regions)
         it++;
     }
 
+    // Check to see if any aggroable mobs should be aggroed by other mobs
+    for (auto PMobIter : m_mobList)
+    {
+        CMobEntity* PMob = dynamic_cast<CMobEntity*>(PMobIter.second);
+        if (PMob == nullptr || PMob->allegiance != ALLEGIANCE_TYPE::PLAYER || !PMob->m_isAggroable)
+        {
+            continue;
+        }
+
+        for (auto PMobCurrentIter : m_mobList)
```

Comment:
Do we really need to iterate over m_moblist twice, or could we apply the continue checks in one?

Reply:
I'm not sure how we could avoid iterating over it twice unless we did something like I mentioned in the PR where we would introduce spatial hashing to quickly retrieve nearby mobs and only iterate over those. All of the conditional checks in the inner for loop are necessary as they use `PCurrentMob`.

Reply:
I guess I'm not understanding, as the outer and inner loop appear to be iterating over the same thing, except for the fact that the outer is continuing on some instances

Reply:
I'll break it down and if desired I can add in comments into the code. The outer loop iterates over all mobs looking for any mobs that are aggroable. If we find such a mob then we iterate over all other mobs looking to see if one of those inner loop mobs can aggro the mob from the outer loop.

Reply:
One way we can avoid this is to work in reverse

When we iterate all the mobs for updates on the main loops above, we can build a list of all the mobs who are aggro able. Then down here we can iterate over them and then ask the zone mob list to try and aggro them, unrolling this loop

Reply:
(but also +1 for some kind of spacial hashing, I know Eden split up a zone into 50" squares and only checks current and adjacent squares, but I'm also partial to quadtrees, which I'd love to implement)

Reply:
Collecting aggroable mobs into a vector is a good idea. I went ahead and implemented that.

You're welcome to do the quadtree or whatever anytime you want. I've implemented spatial hashes before but haven't done it with quadtrees. I'm not sure what advantages you'd get over quadtrees as I think most if not all cells will be filled. Maybe if we are doing something like collision with them as well and wanted something smaller than 50".

Reply:
I guess you're right, most of the saving will be splitting things out from the 50' draw distance. You can go ahead with the spatial hashing if you'd like 👍